src/hashing.rs

```diff
@@ -49,6 +48,26 @@ use crate::{
     keys::SecretKey,
 };
 
+/// An enumeration specifying flags used by hashers.
+/// Flags prepend anything included as hasher input.
+///
+/// The idea is that when putting input into a hasher, we include several things:
+/// - A flag indicating the input type, encoded as a byte
+/// - If the input is of variable length (determined by the type), a little-endian 64-bit encoding of the length
+/// - The input, encoded as bytes
+/// By doing so, we mitigate the risk of collision.
+#[repr(u8)]
+enum Flag {
+    /// An initial domain separator indicating the purpose of the hasher
+    DomainSeparator = 0,
+    /// The version of the hasher, which MUST be a single byte
+    Version,
+    /// A label that can be used to differentiate uses of the hasher
+    Label,
+    /// Arbitrary byte data to be added to the hasher
+    Data,
```

Comment:
I wouldn't know when to use data and when to use label. Perhaps we only need one of them?

Reply:
I'll update the documentation to make this distinction more clear.